### ISSUE TITLE
Add a font option for the editor, maybe the whole app

### ISSUE THREAD

**Issue by drw158** [Automattic/simplenote-electron#274]
I can see how this might be useful. We could at least add a monospace font option, but I think 3 font options would "feel" better.

- Default
- Monospace
- Serif?

**Comment by JokerQyou** [comment]
Although I think the ultimate solution to this would be allowing users to select any font installed on their computer, starting from basic font configuration might be a good idea. Currently CJK characters looks awful in Simplenote. 

Here is how it looks like when I have default font size:
![image](https://cloud.githubusercontent.com/assets/1465267/15809693/674bd32a-2bc8-11e6-9a6c-80f41e954f57.png)

And here is how it looks like when I manually set it bigger:
![image](https://cloud.githubusercontent.com/assets/1465267/15809697/70b8e07e-2bc8-11e6-83e3-a020b991e361.png)

Basically the only acceptable font rendering is the menu text. ASCII characters are still good, though.

**Comment by kriskarkoski** [comment]
Monospace or alternative font options requested by a user in #2711658-t

**Comment by JeffByers** [comment]
For me, the lack of a mono-spaced or non-proportional font such as Courier New, or Consolas on all Simplenote supported platforms is a deal breaker, it's a requirement. I have many notes with column aligned table data, which without the right font, simply does not line up.

**Comment by iamlazynic** [comment]
Really want a mono-space option :+1:

**Comment by nemonein** [comment]
I hope you do something for Korean users.
![simplenote_001](https://cloud.githubusercontent.com/assets/5688288/26409790/8839899a-40dc-11e7-84a2-7ca017ca955d.png)
All Korean letters are displayed as squares.
Those are the notes I created on other platforms.(Android or Windows)

It's snap installed Simplenote 1.0.8 on Ubuntu 17.04.

**Comment by gerv** [comment]
I would expect that to be because the hard-coded font doesn't have Korean glyphs. So yes, this bug would fix that, as would implementing some sort of fallback mechanism (e.g. a set of fonts in CSS).

**Comment by drw158** [comment]
@nemonein @gerv We don't use any custom fonts, and we only use system fonts. It should fallback to a font on your computer with the correct glyphs in your language.

See our font-stack here: https://github.com/Automattic/simplenote-electron/blob/master/scss/variables.scss#L30

**Comment by nemonein** [comment]
@drw158 Thanks for your replay.
I would open a new issue, but I want to make sure something.
I'm not a programmer, so I can't read the codes well.

I just saw the codes you linked, but I'm not sure exactly what font is used for Ubuntu flavours.
Is it sans? or Ubuntu?

If it's Ubuntu, it does not have Korean glyphs.
If it's sans, it is normally matched to Ubuntu fonts in Ubuntu, which results in the same consequences(No Korean fonts).

I don't think Simplenote uses 'sans' because I manually matched 'sans' with Korean font.
```fc-match sans
NanumBarunGothic.ttf: "나눔바른고딕" "Regular"
```
NanumBarunGothic is Korean font, which also has Latin characters.

**Comment by drw158** [comment]
@nemonein Sure, no problem.

Simplenote will try to load the fonts in a sequence from left to right. For example, it will try to find the first font, "-apple-system", and if it doesn't find it, it will look for the next font in the list, "BlinkMacSystemFont". Since your computer has the "Ubuntu" font, it loads that one.

**Comment by gerv** [comment]
But does electron do font substitution if a glyph isn't found, like a browser does? If not, then this could be a problem, because the first font found will be used, and if it doesn't have the right glyphs, you are outta luck.

**Comment by drw158** [comment]
Let's create a new issue and discuss further.

**Comment by nemonein** [comment]
@drw158 @gerv I opened a new issue. #561

**Comment by kylehotchkiss** [comment]
Monospace font would be very nice. My usecase: using as decorators/headlines like I do in my code editor 🙃 

[Source code pro would be awesome](https://github.com/adobe-fonts/source-code-pro) / is open source.

**Comment by silmerusse** [comment]
As a workaround, especially for CJK users, edit Simplenote\resources\app\dist\app.js. Search the font style string (-apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen-Sans", "Ubuntu", "Cantarell", "Helvetica Neue", sans-serif;), and add or replace with your custom font name, for example 'Microsoft YaHei' .

**Comment by zusamann** [comment]
Would suggest the following open source options for Monospace:

https://github.com/tonsky/FiraCode  (ligatures!!)
![image](https://user-images.githubusercontent.com/12589797/31234417-7cd089c0-aa0c-11e7-840a-6637270ca7bf.png)
----
http://madmalik.github.io/mononoki/
![image](https://user-images.githubusercontent.com/12589797/31234389-6eab7198-aa0c-11e7-9643-dc07cfbc816e.png)

**Comment by w3sip** [comment]
Lack of monospace font option is pretty much the only thing preventing me from abandoning Evernote and going with Simplenote instead.

**Comment by mizejewski** [comment]
Please add a monospace font option. This is definitely a dealbreaker and keeps me using https://laverna.cc/ even though I prefer other features of Simplenote.

**Comment by valhristov** [comment]
I started to use Simplenote today and I like it a lot so far. I just found that I can create monospaced inline and multiline blocks with markdown, but having all monospaced text in the editor would be even more awesome.

**Comment by jmillr** [comment]
I would like to join the chorus of voices requesting a mono-spaced option. I, too, would like to migrate my notes from Evernote to Simplenote, but I can't have any 'smart-quotes' messing up my terminal commands!!! Thanks, in advance, if you implement this. [Using MacOS X.]

**Comment by colinhowells** [comment]
Big fan of [Font Bureau's Input Mono](http://input.fontbureau.com/preview/?size=14&language=javascript&theme=monokai&family=InputMono&width=300&weight=300&line-height=1.2&a=0&g=0&i=0&l=0&zero=0&asterisk=0&braces=0&preset=default&customize=please)

Not sure what the license + payment situation would be for inclusion in an app, though

**Comment by yxlao** [comment]
Adding a monospace font option would be super helpful for developers (e.g. code snippets and command lines and etc.).

**Comment by Abaxial** [comment]
Yes please, am very much looking forward to a monospaced option.

**Comment by katinthehatsite** [comment]
Also, requested by a customer in 1042810-zen

**Comment by dadaphl** [comment]
YES! please add a monospace option!

I'm using simplenote as a calendar. For every month I display the output of unix `cal` command. This only works properly with a monospaced font.
```
Su Mo Tu We Th Fr Sa
                1  2
 3  4  5  6  7  8  9
10 11 12 13 14 15 16
17 18 19 20 21 22 23
24 25 26 27 28 29 30
```

**Comment by juanmeleiro** [comment]
Up. I see this issue has been added as an future feature over two years ago. Monospaced fonts are not something to be taken lightly. Monospaced is a matter of mental health.

**Comment by louisabraham** [comment]
I'm using nvalt on macOS but my ASCII diagrams don't render on my phone :/

**Comment by i5o** [comment]
Still waiting I guess

**Comment by sandymcfadden** [comment]
Adding another request from a user here 1576555-zen

**Comment by ernstki** [comment]
Just an update to @silmerusse's suggestion for 2019…

The assets for the desktop app are now packaged with [asar](https://github.com/electron/asar) (a `tar` alternative written in JavaScript). But you can still extract the source files, modify the app's CSS styles, and repackage them with a little bit of effort. For a Linux distribution using the Simplenote `.deb` package, for example:

```bash
# refer to https://stackoverflow.com/a/39373730/785213 if you need
# help setting up your PATH to allow you to run Node packages' CLIs
npm install -g asar

# you will need to give your user permission to modify this directory
cd /opt/Simplenote/resources
sudo setfacl -m u:$USER:rwx .
sudo setfacl -m u:$USER:rw *.asar

# back up the original resource bundle, in case you mess it up
cp app.asar app.asar.factory

# extract the packed resource bundle
asar extract app.asar app

# replace the "-apple-system" font with "Liberation Mono," available
# on most Linux distros, keeping a backup
sed -i.bak s'/-apple-system/"Liberation Mono"/g' app/dist/main.*.css

# optionally, re-pack resource bundle and remove the (temporary) 'app' dir.
# asar pack app app.asar && rm -rf app

# run Simplenote!
simplenote
```

The `-i` option to sed is non-standard, so on a Mac, you will probably need a space between the `-i` and the `.bak` parts.

This is a very blunt approach that replaces _all_ UI fonts with a monospace family, but if you're up for it, you could theoretically target `.note-detail-textarea` separately from `.note-detail-markdown` in the same CSS file.

These steps, unfortunately, will need to be repeated after every Simplenote update.

**Update, 2025**: instructions that work with the new Monaco editor component can be found in [#3036](https://github.com/Automattic/simplenote-electron/issues/3036#issuecomment-3507474583)

**Comment by iamgabrielma** [comment]
Requested as well in 1857067-zen

**Comment by iamgabrielma** [comment]
Another request on 1867947-zen

**Comment by hanchang** [comment]
Another +1 for supporting monospaced fonts please, would immediately switch from Evernote to Simplenote as soon as this is possible!

What are the steps required to do so, in case some intrepid soul wants to take a shot at this?

**Comment by fhamm** [comment]
Please add this feature!

**Comment by reginabally** [comment]
Received another request on 1966461-zen.

**Comment by hoishing** [comment]
\+ 1 for monospace. I saw the comments from an [old post](https://simplenote.com/2017/12/22/big-macos-app-update-available-now/) that it would add monospace support soon, don't know why it didn't implement it till now..

btw, the work around from @ernstki works fine for me. Note that I need to use the [simplenote eletron version](https://github.com/Automattic/simplenote-electron/releases) in macOS, instead of the native app from mac app store, for using the work around.

**Comment by iamgabrielma** [comment]
2041156-zen

**Comment by lcb931023** [comment]
+1 for monospace!

**Comment by amirography** [comment]
+1 for custom font option (or styling in general)

**Comment by kerrypnx** [comment]
+1 for custom font option (or styling in general)

**Comment by iamgabrielma** [comment]
Requested on 2193257-zen

> I am a Chinese writer and the Chinese characters of windows built-in Fonts are so much difficult to read, and I can't change the fonts, which is disappointing and stops me from keeping using Simplenote on Windows...

**Comment by dhry** [comment]
+1 for custom font option. Monospace would be nice, but I'd prefer instead to be allowed to use any font installed on my desktop.

**Comment by cramdesign** [comment]
+1 for Monospace. It may seem minor but it's a requirement for how I organize notes.

**Comment by zhdanjj** [comment]
+1 for Monospace

**Comment by positvt** [comment]
+1 for Monospace, please.
Thank you!

**Comment by lost2** [comment]
+1 for Monospace, please.
Thank you!

**Comment by erolsk8** [comment]
+1 for Monospace

**Comment by ottov1** [comment]
+1 for Monospace. Thanks!

**Comment by spenat28** [comment]
Why was this removed from milestone? Will this be an option in near future?
It’s really needed to have monospaced option at least.

**Comment by codebykat** [comment]
Hi @spenat28, it was removed from the "Future" milestone because we have deprecated that milestone. Everything not in a specific version milestone is on our radar but not specifically slated for an upcoming version.

In other words we've logged the feature request but it is not on our immediate roadmap. Thanks!

**Comment by DesiQuintans** [comment]
I have been waiting for native monospace in Simplenote for years.

**Comment by lost2** [comment]
Is it so difficult to provide mono font option that users are waiting for years now?

**Comment by erol-bm** [comment]
I ended up switching to dropbox (for sync), 1Writer (as iOS editor) and Sublime Text (as desktop editor).

Its working great so far.

**Comment by adyomin** [comment]
@codebykat any update on this? I would love to see monospace custom font support in Simplenote!

**Comment by codebykat** [comment]
Thanks for the continued interest in this! It really does help us gauge what features to prioritize in future.

As you might have seen from recent PRs, @sandymcfadden has been doing a ton of work cleaning up and centralizing the CSS throughout the app, with an eye toward being able to support more themes in future. This would include different fonts. However there's quite a lot of underlying work to be done before we can support this, especially since we would want to do it in the mobile apps as well. The team has been quite busy working on some backend priorities so I don't have any timelines to share as of yet.

For anyone on this thread who doesn't know about it, I did want to mention that any text within code blocks is monospaced when you switch to Markdown mode. I know that doesn't answer all use-cases but it might be a help depending on your needs!

<img width="546" alt="Screen Shot 2021-08-04 at 3 59 28 PM" src="https://user-images.githubusercontent.com/52152/128246794-ef51220e-9e6a-4e61-8dde-bd4684b2299d.png">

<img width="601" alt="Screen Shot 2021-08-04 at 3 59 31 PM" src="https://user-images.githubusercontent.com/52152/128246811-41e0b874-85b6-4f92-ad16-93dc13435611.png">

**Comment by lost2** [comment]
Nice to see mono font have not been forgotten. I guess a lot of user are
still waiting for it.

Thanks a lot

**Comment by DesiQuintans** [comment]
I made a User Style to turn the editor monospace in the meantime. It works with the Stylish/Stylus browser extensions. https://userstyles.org/styles/206552

The userstyles.org website seems to be having some trouble, so here's the CSS:

```
@-moz-document domain("app.simplenote.com")
{
  .view-lines
  {
    font-family: Hack, Monaco, Lucida Console, Courier New, "Simplenote Tasks", monospace !important;
    /* "Simplenote Tasks" styles the checkboxes in Simplenote's checkbox lists. */
}
}
```

![image](https://user-images.githubusercontent.com/3200601/128436693-4b73a9cf-eb49-4072-9c10-d7d2ca4f01a9.png)

**Comment by codebykat** [comment]
Very cool!

**Comment by hobocamp** [comment]
Using other helpful posts above, here's how I was able to change the default font on the native Windows app (current as of this writing, YMMV) (note/warning: changes will revert with each new version you install):

Prerequisites:
- This won't work if you installed Simplenote via Microsoft Store (at least I couldn't figure it out), you'll need to Add/Remove Programs -> Uninstall Simplenote, then go grab & install the latest `Simplenote-win-[version].exe` from [Releases](https://github.com/Automattic/simplenote-electron/releases)
- You'll need a way to extract/create `asar` archive files.  Since I already had [7-Zip](https://www.7-zip.org/), I used [plugin](https://www.tc4shell.com/en/7zip/asar/).  Looks like there's also [WinAsar](https://github.com/aardio/WinAsar), or `npm install -g asar` to use the command line tool.

Steps:
- Close the Simplenote app if it's not already
- Open up/cd to your Simplenote installation directory.  This dir depends on how you installed it, but by default (installed under the current user) it should be `%LocalAppData%\Programs\Simplenote`
- Go into the `resources` dir then, using whatever `asar` app you have, extract `app.asar` to an `app\` subdir
- Once extracted, rename `app.asar` to something else, e.g. `app.asar.bak`
- Under `app\dist\` find the file named "`main.[randomchars].css`" and open that in your preferred plain text/code editor
- Go to the very bottom and append a line:
    `.view-lines { font: 10pt "My Preferred Font" !important; }`
    (obviously change `My Preferred Font` to whatever font name you want, and `10pt` to preferred point size.  If unfamiliar, you can read more about how `font` CSS works on [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/font))
- Save the file to disk, open up Simplenote.  If done correctly (and assuming the app hasn't radically changed), the editor font should change (after short blip.. I couldn't figure out how to fix that).
 
If you notice an issue with the word spacing changing when you highlight text like I did, you can fix that by _also_ appending:
`.view-lines .mtkz { width: auto !important; }`

Advanced users can use Simplenote's Help -> Advanced -> Debugging Console (aka chromium devtools) to find specific styles they want to overwrite.  See also [Hacking Electron Apps](https://www.codepicky.com/hacking-electron-restyle-skype/) tutorial for further tips & tricks.

In the meantime we'll just have to wait for a Custom Font option.  If any of the devs are still out there, if that option is too hard, maybe you could look into something akin to FF's userChrome.css, an external hard-coded .css filename that loads automatically (only if it exists) after all the other stylesheets.

**Comment by ernstki** [comment]
Just an update to [this method](#issuecomment-451258437) for 2023, incorporating @hobocamp's approach of targeting `.view-lines`, which I think is the better way to go.

The below example is for the Simplenote `.deb` package, but could be applied to the tarball as well, _possibly_ even the macOS app bundle (although altering the contents of a signed app seems less and less likely to work these days). On Windows, just follow @hobocamp's [existing instructions above](#issuecomment-1159304416).

```bash
# refer to https://stackoverflow.com/a/39373730/785213 if you need
# help setting up your PATH to allow you to run Node packages' CLIs
npm install -g asar

# you will need to give your user permission to modify this directory
cd /opt/Simplenote/resources
sudo setfacl -m u:$USER:rwx .
sudo setfacl -m u:$USER:rw *.asar

# extract the packed resource bundle
asar extract app.asar app

# specify "Liberation Mono" (available on most Linux distros) for the editor,
# keeping a backup; replace this with whatever size and font family you like
sed -i.bak '$a\
.view-lines { font: 12pt "Liberation Mono" !important; }\
.view-lines .mtkz { width: auto !important; }
' app/dist/main.*.css

# keep the original resource bundle around in case you mess something up
mv app.asar app.asar.factory

# run Simplenote!
simplenote
```

The `-i` option to sed is non-standard, so on a Mac, you would probably need a space between the `-i` and the `.bak` parts.

These steps, unfortunately, will need to be repeated after every Simplenote update. You could copy-paste the relevant commands above into a script somewhere, to make it easier to repeat the steps later.
